torch/testing/_internal/common_utils.py

```diff
@@ -817,17 +819,44 @@ def settings(*args, **kwargs):
     print('Fail to import hypothesis in common_utils, tests are not derandomized')
 
 
+def fetch_and_cache(name: str, url: str) -> str:
+    """
+    Some tests run in a different process so globals like `slow_test_dict` won't
+    always be filled even though the test file was already downloaded on this
+    machine, so cache it on disk
+    """
+    path = os.path.join(tempfile.gettempdir(), name)
+
+    def is_cached_file_valid():
+        # Check if the file is new enough (say 1 hour for now). A real check
+        # could make a HEAD request and check/store the file's ETag
+        fname = pathlib.Path(path)
+        now = datetime.datetime.now()
+        mtime = datetime.datetime.fromtimestamp(fname.stat().st_mtime)
+        diff = now - mtime
+        return diff.total_seconds() < datetime.timedelta(hours=1).seconds
+
+    if os.path.exists(path) and is_cached_file_valid():
+        # Another test process already downloaded the file, so don't re-do it
+        with open(path, "r") as f:
+            return json.load(f)
+    try:
+        contents = urlopen(url, timeout=1).read().decode('utf-8')
+        with open(path, "w") as f:
+            f.write(contents)
+        return json.loads(contents)
+    except Exception as e:
+        print(f'Could not download {url} because of error {e}.')
+        return {}
+
+
 slow_tests_dict: Optional[Dict[str, float]] = None
 def check_slow_test_from_stats(test):
     global slow_tests_dict
     if slow_tests_dict is None:
-        url = 'https://raw.githubusercontent.com/pytorch/test-infra/master/stats/.pytorch-slow-tests'
-        try:
-            contents = urlopen(url, timeout=1).read().decode('utf-8')
-            slow_tests_dict = json.loads(contents)
-        except Exception as e:
-            print(f'Could not download slow test stats because of error {e}. Proceeding with no added slow tests.')
-            slow_tests_dict = {}
+        url = "https://raw.githubusercontent.com/pytorch/test-infra/master/stats/.pytorch-slow-tests"
```

Comment:
This url recently changed to https://raw.githubusercontent.com/pytorch/test-infra/master/stats/slow-tests.json, just fyi